queries.sql

```diff
@@ -31,3 +31,56 @@ WHERE neutered = true;
 SELECT *
 FROM animals
 WHERE  weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+-- Transactions and modifications
+
+BEGIN;
+UPDATE animals SET species = 'undefined';
+SELECT * FROM animals;
+ROLLBACK;
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE animal_name LIKE '%mon';
+
+UPDATE animals SET species = 'pokemon' WHERE animal_name IN ('Charmander', 'Squirtle', 'Blossom', 'Ditto', 'Pikachu');
+COMMIT;
+
+BEGIN;
+DELETE FROM animals WHERE id IN (1,2,3,4,5,6,7,8,9,10,11);
+ROLLBACK;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT my_savepoint;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO SAVEPOINT my_savepoint;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+
+-- AGGREGATES
+
+-- How many animals are there?
+SELECT COUNT(*) AS number_of_animals FROM animals;
+
+-- How many animals have never tried to escape?
+SELECT COUNT(*) AS zero_escape_attempt FROM animals WHERE escape_attempts = 0;
+
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) AS average_weight_kg FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+SELECT animal_name, SUM (escape_attempts) AS total FROM animals
+GROUP BY animal_name ORDER BY total DESC
+LIMIT 1;
```

Comment:
For this question, we're supposed to find which category of animals escapes the most. By category, I mean neutered or not neutered. Something like the following should do: 
```
  SELECT neutered, MAX(escape_attempts) FROM animals
  GROUP BY neutered;
```
You can use `AVG` instead of `MAX` as well.